### PR TITLE
Reorder testcases

### PR DESCRIPTION
**Pull request by robinyuan1002**
Reorder all the testcases(Lind_project testcases and lind-wasm-test testcases)into five folder, and rename lind-wasm-test to unit-tests.

### CODE REVIEW

File: tests/unit-tests/File_and_Directory/truncate.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+	printf("Truncating %s from %d to %d\n", FILENAME, filesize_inc, filesize); 
+	fflush(NULL);
+	check_success(ftruncate(fd, filesize));
+	check_filesize(filesize);
+
+	close(fd);
+
+	return 0;
+}
```
Comment (Yaxuan-w) [inline tests/unit-tests/File_and_Directory/truncate.c:81]: Same question

File: tests/unit-tests/File_and_Directory/writeloop.c [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+int main(void)
+{
+	char str[] = "write succeeded\n";
+	
+	for (int i = 0; i < strlen(str); i++)
+	{
+		write(STDOUT_FILENO, str + i, 1);
+	}
+	return 0;
+}
```
Comment (Yaxuan-w) [inline tests/unit-tests/File_and_Directory/writeloop.c:20]: Same question

File: tests/unit-tests/Process_and_Thread_Management/thread/lind.metadata
Comment (Yaxuan-w) [inline tests/unit-tests/Process_and_Thread_Management/thread/lind.metadata:1]: Do we need to include this?

File: tests/unit-tests/File_and_Directory/chdir_getcwd.c [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  //---------------------------
+  result = getcwd(cwd, sizeof(cwd));
+
+  if (result == NULL)
+    perror("getcwd() error");
+  else
+    printf("current working directory is: %s :: %s\n", cwd, result);
+
+  return 0;
+}
```
[the remark this answers is not in the document]
Comment (robinyuan1002) [reply tests/unit-tests/File_and_Directory/chdir_getcwd.c:27]: I checked the local files have new line at the end but I am not sure why it doesn't appear on github.

File: tests/unit-tests/memory_tests/dup.c
Comment (Yaxuan-w) [inline tests/unit-tests/memory_tests/dup.c:1]: dup/dup2/dupwrite/doubleclose/ioctl should be in file_tests folder

File: tests/unit-tests/process_tests/thread/thread-test
Comment (Yaxuan-w) [inline tests/unit-tests/process_tests/thread/thread-test:1]: should remove

**Review by Yaxuan-w — APPROVED** [review]
looks good now

**Comment by Yaxuan-w** [comment]
Some files in develop branch of lind-nacl still missing. Would be great to add in this repo as well

**Review by Yaxuan-w — APPROVED** [review]
lgtm

**Review by rennergade — APPROVED** [review]

**Pull request merged**